### ISSUE TITLE
Declare intent filter to open a post

### ISSUE THREAD

**Issue by keunes** [tuskyapp/Tusky#3321]
There's a new idea for podcasts: [cross-app comments](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#social-interact) based on ActivityPub. TL:DR – each episode (of podcasts taking part) has a 'root' post/object ([example](https://podcast.thelinuxexp.com/@tlenewspodcast/posts/3be3bb49-690d-4deb-84e7-66da3c4567df)). Replies can be posted to it as usual.

In AntennaPod, the open source podcast player for Android, we would like to make it easy to access & join listener conversations. But rather than including a full interface to view and post comments, the idea is to let dedicated apps handle this. Concretely: from AntennaPod you hit a button, which will open an episode's root post in your favourite ActivityPub/Mastodon app. You then have all the great features you're used to. And if you don't have such app installed, we'll display a short explanation and guide you to some options.

We could just link to the root post URL, but that's [not](https://issuetracker.google.com/issues/258270055) guaranteed to open the right app if installed. And we wouldn't be able to give the user an explanation if such app is not installed.

So, we need an alternative. As I understand, [declaring an intent filter](https://developer.android.com/guide/components/intents-filters#Receiving) is the best alternative. AntennaPod can then send an [implicit intent](https://developer.android.com/guide/components/intents-filters#ExampleSend) in order to open a post. For example: web+activitypub://post/${encoded post uri}

See also
* the initial post announcing the idea: https://fosstodon.org/@AntennaPod/109844984567859264
* the same request with discussion points & suggestions over at Fedilab: https://codeberg.org/tom79/Fedilab/issues/792
* previous ideas about intents: https://github.com/activitypub-schema/proposal/blob/main/PROPOSAL.md & https://github.com/fedi-to/fedi-to.github.io/blob/main/webap.md

Long story short - it would be great if Tusky could implement an intent filter to open posts, so that AntennaPod can call such intent. Curious to hear your thoughts. (Other apps should do the same for this to work well, of course. Discussion with Fedilab is going. Tusky - as one of the bigger Mastodon apps - would be a great contribution.)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by nikclayton** [comment]
https://codeberg.org/tom79/Fedilab/issues/792#issuecomment-804812
